envoy/stream_info/stream_info.h

```diff
@@ -491,6 +493,11 @@ class StreamInfo {
    */
   virtual Ssl::ConnectionInfoConstSharedPtr upstreamSslConnection() const PURE;
 
+  /**
+   * @return const Router::RouteConstSharedPtr Get the route selected for this request.
+   */
+  virtual Router::RouteConstSharedPtr route() const PURE;
```

Comment:
Let's remove `routeEntry()` below as it's now redundant.

Reply:
ok, i will remove it, it will be a large change